### PR TITLE
Epipolar lines cherry-picked and cleaned

### PR DESCRIPTION
**Pull request by yosefm**
Just a rebase of #72  to after the ray_tracing merge, with initial cleanup by me. I'm still not entirely satysfied but we can work from here.

### CODE REVIEW

**Comment by alexlib** [comment]
the tests pass on Mac

**Comment by yosefm** [comment]
@alexlib do you know when crd gets x-sorted before the call to to ``find_candidate()``?

**Comment by alexlib** [comment]
https://github.com/OpenPTV/openptv-python/blob/128f4e1762bb9e781918bdfffb2ac3dfc58b3f4c/src_c/corresp.c#L131

**Comment by yosefm** [comment]
Thanks.

**Comment by yosefm** [comment]
You directed me to the wrong file BTW, this code originates in jw_ptv.c, the file you pointed is an experiment of mine. But now I see that wht find_candidate calls ``crd`` is actually the ``geo`` (flat-image) array. Will work from that.

**Comment by yosefm** [comment]
It looks like we can avoid the "is sorted" complication by simply changing the indexing of the pairwise  candidates lists at the beginning of correspondences_4. After that everything should be consistent. The only other use of pnr is to index the pix array through the geo array, so why not index it always directly with p2?

It's a bit of a hairy work but I can dedicate the time until the next experiment to implementing this change and testing it. The payoff is an easier to use and understand find_candidate, and one less special case threaded through several functions to fret about.

File: liboptv/include/epi.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+typedef struct
+{
+  int pnr;
+  double x, y;
+}
+coord_2d;
+	
+
+void epi_mm (double xl, double yl, Calibration *cal1,
```
Comment (yosefm) [inline liboptv/include/epi.h]: In the following 2, media parameters should preferably be passed by reference (pointer).

**Comment by alexlib** [comment]
As far as I recover from the discussion on the mailing list - nobody really knows what to do. I'm fine with any change you suggest.

File: liboptv/include/epi.h (hunk 1 above)
Comment (alexlib) [inline liboptv/include/epi.h]: Shall I do it ?

Comment (yosefm) [inline liboptv/include/epi.h]: Yes please. I'm still looking at the indexing thing (also took a short break for my experiment).

**Comment by yosefm** [comment]
Better suggestion on the is_sorted issue:
it seems like the one use in mousefunction.c that needs p2 instead of j can be easily acommodated by a one line change that would add a layer of indexing through geo[] to mousefunction's cross-drawing. 

So I suggest to always do the ``is_sorted == 1`` behaviour and discard the flag. Is that acceptable to you?

**Comment by alexlib** [comment]
completely

**Comment by yosefm** [comment]
Ok, I think it is ready now. Please test.

**Comment by alexlib** [comment]
I tested it on Mac OS X - works very well.

**Pull request merged**